main/pipeline/pipeline.cc

```diff
@@ -810,6 +810,173 @@ void unpartitionPackageFiles(vector<ast::ParsedFile> &packageFiles, vector<ast::
     }
 }
 
+namespace {
+
+bool isTestExport(const ast::ExpressionPtr &expr) {
+    auto send = ast::cast_tree<ast::Send>(expr);
+    if (!send || send->fun != core::Names::export_() || send->numPosArgs() != 1) {
+        return false;
+    }
+
+    auto sym = ast::cast_tree<ast::UnresolvedConstantLit>(send->getPosArg(0));
+    while (sym) {
+        if (ast::isa_tree<ast::EmptyTree>(sym->scope)) {
+            return sym->cnst == core::Names::Constants::Test();
+        }
+
+        if (auto parent = ast::cast_tree<ast::UnresolvedConstantLit>(sym->scope)) {
+            sym = parent;
+        } else {
+            break;
+        }
+    }
+
+    return false;
+}
+
+ast::ParsedFile copyPackageWithoutTestExports(const core::GlobalState &gs, const ast::ParsedFile &ast) {
+    ENFORCE(ast.file.isPackage(gs));
+
+    ast::ParsedFile result{ast.tree.deepCopy(), ast.file};
+
+    auto root = ast::cast_tree<ast::ClassDef>(result.tree);
+    if (!root || root->rhs.size() != 1) {
+        return result;
+    }
+
+    auto package = ast::cast_tree<ast::ClassDef>(root->rhs.front());
+    if (!package) {
+        return result;
+    }
+
+    auto it = std::remove_if(package->rhs.begin(), package->rhs.end(), isTestExport);
+    package->rhs.erase(it, package->rhs.end());
+
+    return result;
+}
+
+} // namespace
+
+vector<CondensationStratumInfo> condensationStrata(const core::GlobalState &gs, vector<ast::ParsedFile> &packageFiles,
+                                                   absl::Span<core::FileRef> sourceFiles,
+                                                   const options::Options &opts) {
+    Timer timeit(gs.tracer(), "condensationStrata");
+
+    if (!opts.packageDirected) {
+        return vector{CondensationStratumInfo{absl::MakeSpan(packageFiles), sourceFiles}};
+    }
+
+    auto &db = gs.packageDB();
+
+    auto numPackageFiles = packageFiles.size();
+
+    // We move all the package files into a map, so that it's easy to go from package name to package file while
+    // processing the strata of the traversal.
+    UnorderedMap<core::packages::MangledName, ast::ParsedFile> packagesToPackageRb;
+    for (auto &ast : packageFiles) {
+        auto pkgName = db.getPackageNameForFile(ast.file);
+        ENFORCE(pkgName.exists());
+
+        packagesToPackageRb[pkgName] = move(ast);
+    }
+
+    // We'll generate non-test variants of each package, so we need to double the storage of this vector.
+    packageFiles.clear();
+    packageFiles.reserve(numPackageFiles * 2);
+
+    auto traversal = db.condensation().computeTraversal(gs);
+    ENFORCE(!traversal.parallel.empty());
```

Comment:
minor: I'm not sure what `parallel` refers to. It sounds like it should be a boolean but it clearly is a container of some sort. I wonder if there's a different name for this concept, but I know you already shipped that in a previous change. Don't hold up this PR trying to come up with one

Reply:
I think now that we've switched to using strata/stratum elsewhere, we should make that change here too.